### PR TITLE
ci: send notifications to teams channel

### PR DESCRIPTION
**Pull request by miampf**
Add the `post_to_teams` github action that allows a workflow to post into the `Contrast-CI-Notifications` channel. Furthermore, integrate this action into weekly running actions.

### CODE REVIEW

File: .github/actions/post_to_teams/action.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - name: Post to teams
+      shell: bash
+      run: |
+        curl \
+          -H "Content-Type: application/json" \
+          -d '{"workflow_name": "${{github.workflow}}", "job_id": "${{github.job}}", "run_id": "${{github.run_id}}", "ref": "${{github.ref_name}}", "message": "${{inputs.message}}"}' \
```
Comment (burgerdev) [inline .github/actions/post_to_teams/action.yml:25]: Can we quote the message string, or at least document what it may and may not contain?

**Review by burgerdev — APPROVED** [review]

**Comment by katexochen** [comment]
We should use a card like with Constellation and provide a direct link/button to the logs.

**Comment by miampf** [comment]
> We should use a card like with Constellation and provide a direct link/button to the logs.

Cards are now posted in the corresponding channel.

File: .github/actions/post_to_teams/action.yml (hunk 1 above)
Comment (miampf) [reply .github/actions/post_to_teams/action.yml]: Seems like quoted strings just disappear from the message, should I just document this in the description of the workflow run?

Comment (miampf) [reply .github/actions/post_to_teams/action.yml]: Documented this behavior in the description of the `message` parameter.

File: .github/actions/post_to_teams/action.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    required: true
+  message:
+    description: "The message to be sent. Note that you may not use quotes or escaped quotes in your message."
+    required: true
+
+outputs:
+  commit-msg:
+    description: "Commit message"
+    value: ${{ steps.bump-version.outputs.commit-msg }}
+
```
Comment (Freax13) [inline .github/actions/post_to_teams/action.yml:16]:
```suggestion
```

**Comment by katexochen** [comment]
> > We should use a card like with Constellation and provide a direct link/button to the logs.
> 
> Cards are now posted in the corresponding channel.

Button to run seems to be broken?

**Comment by miampf** [comment]
> > > We should use a card like with Constellation and provide a direct link/button to the logs.
> > 
> > 
> > Cards are now posted in the corresponding channel.
> 
> Button to run seems to be broken?

I don't think the buttons are broken, its just that the messages I sent contained quotes which seemingly broke the parsing of the workflow and caused the URL to be replaced with the last word of the quoted text. If you look at the cardwith run_id 11177841920 (which uses the same adaptive card layout) you'll see that the button works there :)

**Review by Freax13 — APPROVED** [review]

**Review by katexochen — APPROVED** [review]

**Pull request merged**